pennylane/plugins/tests/test_gates.py

```diff
@@ -0,0 +1,489 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests that application of gates and state preparations
+works correctly an a device.
+"""
+import pytest
+
+import numpy as np
+import pennylane as qml
+from scipy.linalg import block_diag
+from flaky import flaky
+
+np.random.seed(42)
+
+# ==========================================================
+# Some useful global variables
+
+# non-parametrized qubit gates
+I = np.identity(2)
+X = np.array([[0, 1], [1, 0]])
+Y = np.array([[0, -1j], [1j, 0]])
+Z = np.array([[1, 0], [0, -1]])
+H = np.array([[1, 1], [1, -1]]) / np.sqrt(2)
+S = np.diag([1, 1j])
+T = np.diag([1, np.exp(1j * np.pi / 4)])
+SWAP = np.array([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
+CNOT = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]])
+CZ = np.diag([1, 1, 1, -1])
+toffoli = np.diag([1 for i in range(8)])
+toffoli[6:8, 6:8] = np.array([[0, 1], [1, 0]])
+CSWAP = block_diag(I, I, SWAP)
+
+# parametrized qubit gates
+phase_shift = lambda phi: np.array([[1, 0], [0, np.exp(1j * phi)]])
+rx = lambda theta: np.cos(theta / 2) * I + 1j * np.sin(-theta / 2) * X
+ry = lambda theta: np.cos(theta / 2) * I + 1j * np.sin(-theta / 2) * Y
+rz = lambda theta: np.cos(theta / 2) * I + 1j * np.sin(-theta / 2) * Z
+rot = lambda a, b, c: rz(c) @ (ry(b) @ rz(a))
+crz = lambda theta: np.array(
+    [
+        [1, 0, 0, 0],
+        [0, 1, 0, 0],
+        [0, 0, np.exp(-1j * theta / 2), 0],
+        [0, 0, 0, np.exp(1j * theta / 2)],
+    ]
+)
+
```

Comment:
(blacking may be needed)
```suggestion
cry = lambda theta: np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, math.cos(theta / 2), -math.sin(theta / 2)], [0, 0, math.sin(theta / 2), math.cos(theta / 2)]])
crx = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, math.cos(theta / 2), 1j * math.sin(-theta / 2)], [0, 0, 1j * math.sin(-theta / 2), math.cos(theta / 2)]])

```